### PR TITLE
Adding new rule API_GW_ENDPOINT_TYPE_CHECK

### PR DESCRIPTION
**Pull request by shikharj05**
I confirm these files are made available under CC0 1.0 Universal (https://creativecommons.org/publicdomain/zero/1.0/legalcode)

*Issue #, if available:*

*Description of changes:*

### CODE REVIEW

File: python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+
+Trigger:
+  Configuration Change on AWS::ApiGateway::RestApi
+
+Reports on:
+  AWS::ApiGateway::RestApi
+
+Rule Parameters:
+  endpointConfigurationType
+  (Required) Comma-separated list of allowed endpoint types.
```
Comment (jongogogo) [inline python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py]: Add a sentence on what are the acceptable types

File: python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    if configuration_item['configuration']['endpointConfiguration']['types'][0] in valid_rule_parameters:
+        return build_evaluation_from_config_item(configuration_item, 'COMPLIANT')
+    return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation="This Amazon API Gateway endpoint type does not match as specified in rule parameter(endpointConfigurationType): " + str(valid_rule_parameters) + ".")
```
Comment (jongogogo) [inline python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py]: Change to 
`The Endpoint Type for this API Gateway API does not match the specified rule parameter (endpointConfigurationType):`

File: python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+def evaluate_parameters(rule_parameters):
+    if 'endpointConfigurationType' not in rule_parameters:
+        raise ValueError("endpointConfigurationType is a mandatory parameter.")
+    rule_parameters_list = rule_parameters['endpointConfigurationType'].split(',')
+    allowed_rule_parameter_values = ["REGIONAL", "PRIVATE", "EDGE"]
```
Comment (jongogogo) [inline python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py]: Create a Global variable for that, put it under `# Other parameters (no change needed)`

File: python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    if configuration_item['configuration']['endpointConfiguration']['types'][0] in valid_rule_parameters:
+        return build_evaluation_from_config_item(configuration_item, 'COMPLIANT')
+    return build_evaluation_from_config_item(configuration_item, 'NON_COMPLIANT', annotation="This Amazon API Gateway endpoint type does not match as specified in rule parameter(endpointConfigurationType): " + str(valid_rule_parameters) + ".")
+
+def evaluate_parameters(rule_parameters):
+    if 'endpointConfigurationType' not in rule_parameters:
+        raise ValueError("endpointConfigurationType is a mandatory parameter.")
+    rule_parameters_list = rule_parameters['endpointConfigurationType'].split(',')
```
Comment (jongogogo) [inline python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK.py]: please manage if there is an extra space. You can use strip() somewhere in your code

File: python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK_test.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    def client(client_name, *args, **kwargs):
+        if client_name == 'config':
+            return CONFIG_CLIENT_MOCK
+        if client_name == 'sts':
+            return STS_CLIENT_MOCK
+        raise Exception("Attempting to create an unknown client")
+
+sys.modules['boto3'] = Boto3Mock()
+
+RULE = __import__('API_GW_ENPOINT_TYPE_CHECK')
```
Comment (jongogogo) [inline python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK_test.py]: Typo in "endpoint"

File: python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK_test.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+    def test_no_rule_parameter(self):
+        rule_parameter_empty = '{}'
+        response = RULE.lambda_handler(build_lambda_configurationchange_event(invoking_event=self.invoking_event_regional, rule_parameters=rule_parameter_empty), '{}')
+        assert_customer_error_response(self, response, 'InvalidParameterValueException', "endpointConfigurationType is a mandatory parameter.")
+
+    #Scenario 2: Rule parameter value is invalid
+    def test_invalid_value_rule_parameter(self):
+        rule_parameter_invalid = '{"endpointConfigurationType":"INVALID"}'
+        allowed_rule_parameter_values = ["REGIONAL", "PRIVATE", "EDGE"]
```
Comment (jongogogo) [inline python/API_GW_ENPOINT_TYPE_CHECK/API_GW_ENPOINT_TYPE_CHECK_test.py]: Use a global variable

**Comment by shikharj05** [comment]
> You have a typo in the name of the rule :)

Thank you for pointing it out, I had missed this. I have made the changes. The last 2 commits should include all the suggested changes. Let me know if any other change is required.

**Pull request merged**